transport/http/config.go

```diff
@@ -133,13 +134,28 @@ func (ts *transportSpec) buildInbound(ic *InboundConfig, t transport.Transport,
 // 	      http:
 // 	        url: "http://127.0.0.1:80/"
 type OutboundConfig struct {
+	config.PeerListConfig
+
 	// URL to which requests will be sent for this outbound. This field is
 	// required.
 	URL string `config:"url,interpolate"`
 }
 
 func (ts *transportSpec) buildOutbound(oc *OutboundConfig, t transport.Transport, k *config.Kit) (*Outbound, error) {
-	return t.(*Transport).NewSingleOutbound(oc.URL, ts.OutboundOptions...), nil
+	x := t.(*Transport)
+
+	chooser, err := oc.PeerListConfig.BuildChooser(x, hostport.Identify, k)
+	if err != nil {
+		return nil, fmt.Errorf(`cannot configure peer chooser for HTTP outbound: %v`, err)
+	}
+
+	opts := make([]OutboundOption, len(ts.OutboundOptions))
+	_ = copy(opts, ts.OutboundOptions)
+	if oc.URL != "" {
+		opts = append(opts, URLTemplate(oc.URL))
```

Comment:
You can just do,

    opts := ts.OutboundOptions
    if oc.URL != "" {
      opts = append(opts, URLTemplate(oc.URL))
    }

The original slice won't be modified because it still thinks its length is N
and the new option is out of bounds.

```
.------ts.OutboundOptions------.  
v                              v
+-------+--------+-----+-------+-------------+
| opt 1 | opt  2 | ... | opt N | URLTemplate |
+-------+--------+-----+-------+-------------+
```